scripts/release/release_notes.py

```diff
@@ -0,0 +1,132 @@
+import argparse
+import pathlib
+
+from git import Repo
+from jinja2 import Template
+
+from scripts.release.changelog import (
+    DEFAULT_CHANGELOG_PATH,
+    DEFAULT_INITIAL_GIT_TAG_VERSION,
+    ChangeEntry,
+    ChangeKind,
+    get_changelog_entries,
+)
+from scripts.release.version import (
+    calculate_next_release_version,
+    find_previous_version,
+)
+
+
+def generate_release_notes(
+    repository_path: str = ".",
+    changelog_sub_path: str = DEFAULT_CHANGELOG_PATH,
+    initial_commit_sha: str = None,
+    initial_version: str = DEFAULT_INITIAL_GIT_TAG_VERSION,
+) -> str:
+    f"""Generate a release notes based on the changes since the previous version tag.
+
+    Parameters:
+        repository_path: Path to the Git repository. Default is the current directory '.'.
+        changelog_sub_path: Path to the changelog directory relative to the repository root. Default is '{DEFAULT_CHANGELOG_PATH}.
+        initial_commit_sha: SHA of the initial commit to start from if no previous version tag is found.
+        initial_version: Version to use if no previous version tag is found. Default is '{DEFAULT_INITIAL_GIT_TAG_VERSION}'.
+
+    Returns:
+        Formatted release notes as a string.
+    """
```

Comment:
Let's get rid of default values and always assume we receive values here. We already have defaulting in the arguments: let's not maintain two places where we default.

Reply:
I'm not sure that I follow. Perhaps you did not make it very clear. I'll try again.

I'm suggesting:
1. Keep default values in `ArgumentParser`.
2. Remove default values from `generate_release_notes` function because it duplicates defaulting from `ArgumentParser` and we always call it so we pass values for all the arguments. So defaults here are not really required.

I don't see a similar issue with `calculate_next_version.py`.